compiler-core/src/dependency.rs

```diff
@@ -127,6 +153,7 @@ struct DependencyProvider<'a> {
     packages: RefCell<HashMap<EcoString, hexpm::Package>>,
     remote: Box<dyn PackageFetcher>,
     locked: &'a HashMap<EcoString, Version>,
+    exact_only: &'a HashMap<String, Version>,
```

Comment:
I added a separate map here because I wasn't sure if modifying the way we create locked would make sense/cause issues elsewhere